x/muxer/muxer.go

```diff
@@ -424,24 +424,12 @@ func (m *muxer) GetPluginInfo(ctx context.Context, e *emptypb.Empty) (*rpc.Plugi
 }
 
 func (m *muxer) Attach(ctx context.Context, req *rpc.PluginAttach) (*emptypb.Empty, error) {
-	attach := make([]func() error, len(m.servers))
-	for i, s := range m.servers {
-		s := s
-		attach[i] = func() error {
-			_, err := s.Attach(ctx, req)
-			return err
-		}
-	}
-
-	var closeErr error
-	if m.host != nil {
-		closeErr = m.host.Close()
+	host, err := provider.NewHostClient(req.GetAddress())
+	if err != nil {
+		return nil, err
 	}
-
-	var err error
-	m.host, err = provider.NewHostClient(req.GetAddress())
-
-	return &emptypb.Empty{}, errors.Join(append(asyncJoin(attach), err, closeErr)...)
+	m.host = host
```

Comment:
Since this host client is now not propagated to the constituent providers they will not be able to work correctly (send logs to the host for example).

Reply:
Yup, that was a red herring; see above